eth/vm/logic/arithmetic.py

```diff
@@ -187,12 +184,13 @@ def shl(computation):
     Bitwise left shift
     """
     shift_length, value = computation.stack_pop(num_items=2, type_hint=constants.UINT256)
-    bit_vector = BitVector(intVal=value, size=constants.CONST_256)
 
     if shift_length >= constants.CONST_256:
         result = constants.CONST_0
+    elif shift_length == constants.CONST_0:
+        result = value
     else:
-        result = bit_vector.shift_left(shift_length).int_val()
+        result = (value * 2**shift_length) & constants.UINT_256_MAX
```

Comment:
A quick test shows that replacing this with `(value << shift_length) & constants.UINT_256_MAX` passes all the tests.

Reply:
Indeed! I have to read up on that magic!